src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
```

Comment:
@bording OK, I guess it's better that I try and think of a situation where a user might read that and say "WTF?".

How about:

> I have a few event types, and they all implement `IEvent`, as shown in the samples. Everything works nicely and I deploy my system. When building a new feature, I decide to sprinkle a few of my event types with some new interfaces since it seems to the best way to change things. I smoke test my app and I get a log message telling me that my messages shouldn't implement events. WTF? They've always implemented `IEvent`, so why am I seeing this message for the first time? So you're telling me I'm not supposed to implement interfaces, but your samples all implement from `IEvent`. WTF?